examples/celllistmap.jl

```diff
@@ -0,0 +1,249 @@
+# energy function is a smoothly decaying potential with a maximum when
+# the particles overlap.
+#
+# Thus, if the maximum sum of radii between particles is much smaller than the size
+# of the system, cell lists can greatly accelerate the computation of the pairwise forces.
+#
+# Each particle will have different radii and different repulsion force constants and masses.
+using Agents
+
+# Below we define the `Particle` type, which represents the agents of the
+# simulation. The `Particle` type, for the `ContinousAgent{2}` space, will have additionally
+# an `id` and `pos` (positon) and `vel` (velocity) fields, which are automatically added
+# by the `@agent` macro.
+@agent Particle ContinuousAgent{2} begin
+    r::Float64 # radius
+    k::Float64 # repulsion force constant
+    mass::Float64
+end
+Particle(; id, pos, vel, r, k, mass) = Particle(id, pos, vel, r, k, mass)
+
+# ## Required and data structures for CellListMap.jl
+import CellListMap
+using StaticArrays
+# `StaticArrays` provides the `SVector` type, which is practical for the representation of
+# various vector types (e.g., coordinates or velocities) in small amount of dimensions.
+# Agents.jl uses `NTuple{D, Float64}` for that, which does not support vector operations
+# out of the box. In the future, Agents.jl may also switch the `pos` type to a static vector.
+
+# We `import` the `CellListMap` package because
+# it exports some functions with conflicting names with `Agents`.
+
+# 
+# ## Model properties.
+# 
+# The `clmap_system` field will store the data required for `CellListMap`, and again we 
+# use a parametric type to guarantee the concrectness of the types, aoviding type
+# instabilities.  
+#
+Base.@kwdef struct Properties{CL<:CellListMap.PeriodicSystem}
+    dt::Float64 = 0.01
+    number_of_particles::Int64 = 0
+    clmap_system::CL # CellListMap system
+end
+
+# Two auxiliary arrays will be created on model initialization, to be passed to
+# the `CellListMap.PeriodicSystem` data structure:
+#
+# 1. `positions`: `CellListMap` requires a vector of (preferentially) static vectors as the positions
+#    of the particles. To avoid creating this array on every call, a buffer to
+#    which the `agent.pos` positions will be copied is stored in this data structure.
+# 2. `forces`: In this example, the property to be computed using `CellListMap.jl` is 
+#    the forces between particles, which are stored here in a `Vector{<:SVector}`, of
+#    the same type as the positions. These forces will be updated by the `CellListMap.map_pairwise!`
+#    function.
+#
+# Additionally, the computation with `CellListMap.jl` requires the definition of a `cutoff`,
+# which will be twice the maximum interacting radii of the particles, and the geometry of the
+# the system, given by the `sides` of the periodic box. 
+#
+# Just for clarity, we define an alias for the function that fetches the forces from 
+# the `CellListMap.PeriodicSystem` structure:
+const get_forces = CellListMap.get_output
```

Comment:
I'd advise to not do this and use the full call name instead.

Reply:
Actually, forget what I said. You are right, this makes for a more intuitive name call.